### ISSUE TITLE
Opening up beyond OpenAI / Azure OpenAI?

### ISSUE THREAD

**Issue by ChristianWeyer** [microsoft/semantic-kernel#482]
Are there any plans to offer an open architecture like Langchain where we connect to literally any model, and not just OpenAI / Azure OpenAI?
Using models from Huggingface is such a breeze with Langchain.

Thank you.

**Comment by ChristianWeyer** [comment]
Will the .NET-based SK implementation also get more integrated native access to HF? This would be very helpful for devs. @awharrison-28

**Comment by ChristianWeyer** [comment]
This is also great :-)

However, maybe I am asking for having a ctor on https://github.com/microsoft/semantic-kernel/blob/main/dotnet/src/SemanticKernel/Connectors/HuggingFace/TextEmbedding/HuggingFaceTextEmbeddingGeneration.cs that just takes the model string, and not the Url if we are talking to HF?

BTW: is there a public roadmap?

Thank you!

**Comment by awharrison-28** [comment]
We don't currently have a public roadmap, but this PR is a start to one, check it out :) https://github.com/microsoft/semantic-kernel/pull/509

As for URL vs model string, ML.Net OnnxRuntime are the closest that C# can get to natively and locally getting model inference. There is no native C# support for Hugging Face, so all inference must be collected from an external "endpoint" of some sort, be it localhost or remote service. URL is most universal method for us to support this at the moment. 

Now that we've are officially supporting Semantic Kernel with Python, native integration with Hugging Face will be possible if you're willing to use a different platform. Coming soon.

**Comment by ChristianWeyer** [comment]
In Langchain, we can simply say
`HuggingFaceEmbeddings(model_name="distiluse-base-multilingual-cased-v2")`

which then talks to HF.

This is what I am talking about for the .NET side :-)

**Comment by awharrison-28** [comment]
In short:

We will very shortly have the ability to the equivalent of `HuggingFaceEmbeddings(model_name="distiluse-base-multilingual-cased-v2")` in the [Python SK](https://github.com/microsoft/semantic-kernel/tree/main/python). We are a little ways off from having the same seamless experience with .NET, but we want to get there! 

Context - I apologize if you're already aware of the following:

LC is able to seamlessly communicate with HF because it is dependent on the following python HF packages - `transformers', `sentence-transformers', which in turn are dependent on PyTorch. They provide the plumbing to get the model and inference with it just by providing the model card string. The Python SK will be able to leverage these packages as well (coming very soon). 

We don't have the luxury of being able to rely on a `transformers` or `PyTorch` library from HF in .NET unfortunately. We are exploring creating a seamless HF experience (and other bring your own model types of experiences) using ML.NET, OnnxRuntime, and possibly TorchSharp, but at this time, we have to develop the majority of the end-to-end solution ourselves. 

TLDR; Python gives you a lot for free in the ML/AI world; we have much less to work with in .NET.

**Comment by ChristianWeyer** [comment]
Thank you, that thoroughly answers it :-)
